psi4/src/psi4/libfock/cubature.cc

```diff
@@ -4952,250 +4949,4 @@ void OctreeGridBlocker::block() {
     }
 }
 
-RadialGrid::RadialGrid() : npoints_(0) {}
-RadialGrid::~RadialGrid() {
-    if (npoints_) {
-        delete[] r_;
-        delete[] w_;
-    }
-}
-void RadialGrid::print(std::string out, int level) const {
-    std::shared_ptr<psi::PsiOutStream> printer = (out == "outfile" ? outfile : std::make_shared<PsiOutStream>(out));
-    if (level > 0) {
-        printer->Printf("   => RadialGrid: %s Scheme <=\n\n", scheme_.c_str());
-        printer->Printf("      Points: %d\n", npoints_);
-        printer->Printf("      Alpha:  %24.16E\n\n", alpha_);
-        printer->Printf("   %4s %24s %24s\n", "N", "R", "W");
-        if (level > 1) {
-            for (int i = 0; i < npoints_; i++) {
-                printer->Printf("   %4d %24.16E %24.16E\n", i + 1, r_[i], w_[i]);
-            }
-        }
-        printer->Printf("\n");
-    }
-}
-std::shared_ptr<RadialGrid> RadialGrid::build(const std::string &scheme, int npoints, double alpha, int Z) {
-    if (scheme == "BECKE") {
-        return RadialGrid::build_becke(npoints, alpha, Z);
-    } else if (scheme == "TREUTLER") {
-        return RadialGrid::build_treutler(npoints, alpha, Z);
-    } else {
-        throw PSIEXCEPTION("RadialGrid::build: Unrecognized radial grid.");
-    }
-}
-std::shared_ptr<RadialGrid> RadialGrid::build(const std::string &scheme, int npoints, double *r, double *wr,
-                                              double alpha, int Z) {
-    RadialGrid *grid = new RadialGrid();
-
-    grid->scheme_ = scheme;
-    grid->npoints_ = npoints;
-    grid->alpha_ = alpha;
-    grid->r_ = new double[npoints];
-    grid->w_ = new double[npoints];
-
-    ::memcpy(grid->r_, r, sizeof(double) * npoints);
-    ::memcpy(grid->w_, wr, sizeof(double) * npoints);
-
-    return std::shared_ptr<RadialGrid>(grid);
-}
-std::shared_ptr<RadialGrid> RadialGrid::build_becke(int npoints, double alpha, int Z) {
-    RadialGrid *grid = new RadialGrid();
-
-    grid->scheme_ = "BECKE";
-    grid->npoints_ = npoints;
-    grid->alpha_ = alpha;
-    grid->r_ = new double[npoints];
-    grid->w_ = new double[npoints];
-
-    for (int tau = 1; tau <= npoints; tau++) {
-        double x = cos(tau / (npoints + 1.0) * M_PI);
-        double r = alpha * (1.0 - x) / (1.0 + x);
-        double temp = sin(tau / (npoints + 1.0) * M_PI);
-        double w =
-            M_PI / (npoints + 1.0) * temp * temp * alpha * 2.0 / ((1.0 + x) * (1.0 + x) * sqrt(1.0 - x * x)) * r * r;
-        grid->r_[tau - 1] = r;
-        grid->w_[tau - 1] = w;
-    }
-
-    return std::shared_ptr<RadialGrid>(grid);
-}
-std::shared_ptr<RadialGrid> RadialGrid::build_treutler(int npoints, double alpha, int Z) {
```

Comment:
This is the only Treutler version with correct Eta values. It does give slightly different results compared to the currently used "Treutler" where Eta is always 1. Though I had no time to try to confirm if this one is fully correct.

Reply:
Even if it's correct, it was never actually used anywhere. The only use of the `RadialGrid` built was to provide data in `print_details` -- it's not accessed by anything or anyone outside of that function, at least as far as all of the psi4 tests can determine. In addition, the only calls to RadialGrid::build specifically were of the type `RadialGrid::build("Unknown, ...` which never invoked `build_becke` or `build_treutler`.

Reply:
Yes, i tested it in a local build. So many parts of cubature.cc are messy unfortunately. Could you move the Eta table to line 2045 as a comment and add a note that the eta mapping is missing, please?
When I added the table long time ago I wasn't aware these routines were never called.

Reply:
Done